### PR TITLE
Add prow configs for kube-scheduler-simulator

### PR DESCRIPTION
**Pull request by pravarag**
As part of this PR, we want to enable Prow based CI checks for kubernetes-sigs/kube-scheduler-simulator project.

### CODE REVIEW

File: config/jobs/kubernetes-sigs/kube-scheduler-simulator/OWNERS
Comment (pravarag) [reply config/jobs/kubernetes-sigs/kube-scheduler-simulator/OWNERS]: Updated with the latest push.

**Comment by pravarag** [comment]
I've opened this PR for review @sanposhiho but I had few queries before it's merged:

@BenTheElder @spiffxp @wojtek-t is there any way I can test these configuration changes from my feature branch with prow or, these can only be tested once the PR is merged? I wanted to check if the configuration has all the pieces in place without any errors.

**Comment by BenTheElder** [comment]
you have to merge the configs.
you should probably start with either `optional: true` or also `always_run: false` and manually test out the new jobs before jumping to making them always run + required.

**Comment by pravarag** [comment]
> you have to merge the configs. you should probably start with either `optional: true` or also `always_run: false` and manually test out the new jobs before jumping to making them always run + required.

Thanks @BenTheElder , I've updated the config with `always_run:false` and now it can be merged if there are no other changes. However, I keep getting `pull-test-infra-unit-test` failed. And upon checking I found out the main error as below:

```
Failed
=== FAIL: config/tests/testgrids TestKubernetesProwInstanceJobsMustHaveMatchingTestgridEntries (0.00s)
    config_test.go:533: Job lint-frontend-with-yarn does not have a matching testgrid testgroup
    config_test.go:533: Job pull-kube-scheduler-simulator-get-dependencies does not have a matching testgrid testgroup
    config_test.go:533: Job pull-kube-scheduler-simulator-lint does not have a matching testgrid testgroup
    config_test.go:533: Job pull-kube-scheduler-simulator-run-tests does not have a matching testgrid testgroup 
```
Wanted to check do I need to configure any testgroup for newer entries w.r.t kube-scheduler-simulator?

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
